akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowSlidingSpec.scala

```diff
@@ -3,128 +3,39 @@
-        case _ ⇒ testActor ! "done"
-      }
-      expectMsg(Vector(1, 2, 3))
-      expectMsg(Vector(2, 3, 4))
-      expectMsg(Vector(3, 4, 5))
-      expectMsg(Vector(4, 5, 6))
-      expectMsg(Vector(5, 6, 7))
-      expectMsg("done")
-    }
-
-    "work with n = 3, step = 2" in assertAllStagesStopped {
-      Source(1 to 6).sliding(n = 3, step = 2).runForeach(testActor ! _) onSuccess {
-        case _ ⇒ testActor ! "done"
-      }
-      expectMsg(Vector(1, 2, 3))
-      expectMsg(Vector(3, 4, 5))
-      expectMsg(Vector(5, 6))
-      expectMsg("done")
-    }
-
-    "work with n = 3, step = 2, complete group" in assertAllStagesStopped {
-      Source(1 to 7).sliding(n = 3, step = 2).runForeach(testActor ! _) onSuccess {
-        case _ ⇒ testActor ! "done"
-      }
-      expectMsg(Vector(1, 2, 3))
-      expectMsg(Vector(3, 4, 5))
-      expectMsg(Vector(5, 6, 7))
-      expectMsg("done")
-    }
-
-    "work with n = 3, step = 3" in assertAllStagesStopped {
-      Source(1 to 6).sliding(n = 3, step = 3).runForeach(testActor ! _) onSuccess {
-        case _ ⇒ testActor ! "done"
-      }
-      expectMsg(Vector(1, 2, 3))
-      expectMsg(Vector(4, 5, 6))
-      expectMsg("done")
-    }
-
-    "work with n = 2, step = 3" in assertAllStagesStopped {
-      Source(1 to 6).sliding(n = 2, step = 3).runForeach(testActor ! _) onSuccess {
-        case _ ⇒ testActor ! "done"
-      }
-      expectMsg(Vector(1, 2))
-      expectMsg(Vector(4, 5))
-      expectMsg("done")
-    }
-
-    "work with n = 2, step = 1" in assertAllStagesStopped {
-      Source(1 to 6).sliding(n = 2, step = 1).runForeach(testActor ! _) onSuccess {
-        case _ ⇒ testActor ! "done"
-      }
-      expectMsg(Vector(1, 2))
-      expectMsg(Vector(2, 3))
-      expectMsg(Vector(3, 4))
-      expectMsg(Vector(4, 5))
-      expectMsg(Vector(5, 6))
-      expectMsg("done")
-    }
 
-    "work with n = 3, step = 4" in assertAllStagesStopped {
-      Source(1 to 12).sliding(n = 3, step = 4).runForeach(testActor ! _) onSuccess {
-        case _ ⇒ testActor ! "done"
-      }
-      expectMsg(Vector(1, 2, 3))
-      expectMsg(Vector(5, 6, 7))
-      expectMsg(Vector(9, 10, 11))
-      expectMsg("done")
-    }
+    "behave just like collections sliding" in assertAllStagesStopped {
+      val gen = for {
+        len ← Gen.choose(0, 127)
+        win ← Gen.choose(1, 31)
+        step ← Gen.choose(1, 61)
+      } yield (len, win, step)
 
-    "work with n = 3, step = 6" in assertAllStagesStopped {
-      Source(1 to 12).sliding(n = 3, step = 6).runForeach(testActor ! _) onSuccess {
-        case _ ⇒ testActor ! "done"
+      forAll(gen) {
+        case (len, win, step) ⇒
+          val done = s"done: $len $win $step"
+          val as = Source(() ⇒ Iterator.from(0).take(len)).sliding(win, step)
+          val cs = Source(() ⇒ Iterator.from(0).take(len).sliding(win, step))
+          as.zipWith(cs)(_ should be(_)).runFold(done)(Keep.left) pipeTo testActor
+          expectMsg(done)
```

Comment:
btw, this is what I don't like about ScalaCheck - the list of cases being checked is not visible explicitly. Maybe an IDE plugin to generate test files using ScalaCheck like rules what I would be happy with.